backend/test/index.test.js

```diff
@@ -0,0 +1,51 @@
+const Client = require("socket.io-client");
+const CreateLobbyDTO = require("../domain/DTO/request/CreateLobbyDTO");
+const server = require("../index")
+
+describe("Create-lobby event test", () => {
+    let clientSocket;
+
+    beforeAll((done) => {
+        clientSocket = new Client(`http://localhost:4001`);
+        done();
+    });
+
+    afterAll((done) => {
+        server.close(done);
+    });
+
+    test("Simple Create-lobby event", (done) => {
+        let createLobbyDTO = new CreateLobbyDTO("Anmol");
+        
+        // Subscribe to lobby-code
+        clientSocket.on("lobby-code", (lobbyCodeDTOString) => {
+            let lobbyCodeDTO = JSON.parse(lobbyCodeDTOString)
+            expect(lobbyCodeDTO.code).toBeDefined();
+            done();
+        });
+
+        // Request to create a new lobby
+        clientSocket.emit("create-lobby", JSON.stringify(createLobbyDTO));
+    });
+
+    test("Two hosts with two rooms", (done) => {
+        let createLobbyDTO = new CreateLobbyDTO("Anmol");
+        
+        // Client 1 to subscribe to lobby-code
+        clientSocket.on("lobby-code", (lobbyCodeDTOString) => {
```

Comment:
Since the handler for "lobby-code" is never unsubscribed in the previous test, both handlers are run on every event. The handlers for the "lobby-code" should be unsubscribed when the tests finish.

A solution could be to use the socket.once method rather than socket.on

Reply:
Fixed!!